### PR TITLE
Yandex dataproc deduce default service account

### PR DESCRIPTION
**Pull request by Piatachock**
New features for YandexCloud Provider:
* Support User-Agent header parametrization via environent variable
* Store Service Account ID from Connection in Hook; use it as default in DataprocClusterCreateOperator

### CODE REVIEW

File: airflow/providers/yandex/hooks/yandex.py [hunk 1]
```diff
@@ -85,17 +85,24 @@ def get_connection_form_widgets() -> dict[str, Any]:
             ),
         }
 
+    @classmethod
+    def airflow_user_agent(cls) -> str:
+        import airflow
+        import os
+
+        version = os.getenv("AIRFLOW_USER_AGENT_VERSION", airflow.__version__)
```
Comment (Taragolis) [inline airflow/providers/yandex/hooks/yandex.py]: Maybe better make it as a part of Provider Configuration?
- [Setting Configuration Options](https://airflow.apache.org/docs/apache-airflow/stable/howto/set-config.html)
- [Global Configuration Reference](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#configuration-reference)
- [OpenLinage Configuration Reference](https://airflow.apache.org/docs/apache-airflow-providers-openlineage/stable/configurations-ref.html)

**Comment by eladkal** [comment]
Static checks are failing

File: airflow/providers/yandex/provider.yaml [hunk 2]
```diff
@@ -70,3 +70,15 @@ hooks:
...
     connection-type: yandexcloud
+
+config:
+  airflow:
+    description: ~
+    options:
+      sdk_user_agent_prefix:
+        description: |
+          Prefix for User-Agent header in Yandex.Cloud SDK requests
+        version_added: ~
```
Comment (Taragolis) [inline airflow/providers/yandex/provider.yaml]:
```suggestion
        version_added: 3.6.0
```

File: airflow/providers/yandex/provider.yaml [hunk 3]
```diff
@@ -70,3 +70,15 @@ hooks:
 connection-types:
   - hook-class-name: airflow.providers.yandex.hooks.yandex.YandexCloudBaseHook
     connection-type: yandexcloud
+
+config:
+  airflow:
```
Comment (Taragolis) [inline airflow/providers/yandex/provider.yaml]:
```suggestion
  yandex:
```

File: airflow/providers/yandex/provider.yaml [hunk 4]
```diff
@@ -70,3 +70,15 @@ hooks:
...
+    description: ~
```
Comment (Taragolis) [inline airflow/providers/yandex/provider.yaml]:
```suggestion
    description: This section contains settings for Yandex Cloud integration.
```

or more appropriate

File: airflow/providers/yandex/hooks/yandex.py [hunk 5]
```diff
@@ -88,14 +88,19 @@ def get_connection_form_widgets() -> dict[str, Any]:
...
+        from airflow.configuration import conf
         from airflow.providers_manager import ProvidersManager
 
         try:
             manager = ProvidersManager()
             provider_name = manager.hooks[cls.conn_type].package_name  # type: ignore[union-attr]
             provider = manager.providers[provider_name]
-            return f"apache-airflow/{airflow.__version__} {provider_name}/{provider.version}"
+            return ' '.join((
+                conf.get("yandex", "sdk_user_agent_prefix"),
```
Comment (Taragolis) [inline airflow/providers/yandex/hooks/yandex.py]:
```suggestion
                conf.get("yandex", "sdk_user_agent_prefix", fallback=""),
```

**Comment by Piatachock** [comment]
> Static checks are failing

Sorry, forgot to mark this PR as WIP.
I'm actually trying to build provider package locally to check it in our Managed Airflow installation end-to-end.
Do you, by any chance, know where should I look for any info on the subject? I've found `dev` folder and relevant scripts, but still struggling atm.

**Comment by Taragolis** [comment]
> it in our Managed Airflow 

BTW, you or some one from the Yandex also could add link to Yandex Managed Service for Apache Airflow into [Airflow Ecosystem](https://airflow.apache.org/ecosystem/#airflow-as-a-service), this placed into another repo, however changes could be done by simple click on `Suggest a change on this page` and make PR changes into the Github Editor

The only one thing that I found that International Link to Yandex's managed service https://cloud.yandex.com/en/services/managed-airflow right now return 404

**Comment by Taragolis** [comment]
> I'm actually trying to build provider package locally to check it in our Managed Airflow installation end-to-end.
Do you, by any chance, know where should I look for any info on the subject? I've found dev folder and relevant scripts, but still struggling atm.

You could run this [`breeze`](https://github.com/apache/airflow/blob/main/BREEZE.rst) command

```console
breeze release-management prepare-provider-packages yandex --version-suffix-for-pypi dev0 --package-format both
```

This would create two distribution (wheel, source) in `dist` directory

```console

Good version of Docker: 24.0.6.
Good version of docker-compose: 2.22.0

...

Prepared provider package yandex in format both
===================================================================================

Summary of prepared packages:

    Prepared:
yandex

===================================================================================
apache-airflow-providers-yandex-3.5.0.dev0.tar.gz
apache_airflow_providers_yandex-3.5.0.dev0-py3-none-any.whl
===================================================================================

All good! Airflow packages are prepared in dist folder

```

**Comment by Piatachock** [comment]
Thanks! Managed to build provider locally and test it on our managed solution. Works as expected

**Comment by Taragolis** [comment]
Need to add test for this changes.

For validate configuration, you might use `conf_vars` - context manager / decorator. Some samples:

https://github.com/apache/airflow/blob/eb450b95fdb29b4c1252c2a1ccb703ac397a2de8/tests/always/test_secrets_local_filesystem.py#L431-L446

https://github.com/apache/airflow/blob/1aa91a482630f389005a9a36b0f038e675163411/tests/jobs/test_base_job.py#L113-L123

**Review by Taragolis — APPROVED** [review]

**Comment by Piatachock** [comment]
Andrey, thanks for your approval! Is there anything else that should be done to merge this PR?

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Nope. It looks cool.

**Comment by potiuk** [comment]
Good to see this "per-provider" configuration feature... But that reminds me that we need a bit more "discovery" 

Look at "docs/apache-airflow-providers-celery" - there is configurtions-ref.rst which you have to add to yandex and you will have to also add configuration to index.rst @Piatachock also you should add link to  https://github.com/apache/airflow/blob/main/docs/apache-airflow/configurations-ref.rst

That reminds me that we should add pre-commit and automation - this all should happen semi-automatically or at least should be verified by pre-commit (but it does not yet).

**Pull request merged**